[CONTENT] Application Manager RAPPS GSoC 2020 (#65)
reactos/web-content: content/blogs/GSoC-2020-rapps.md

## How I meet ReactOS and GSoC

*This part is not related to the work I'm going to do, just some story between me and ReactOS*

Less than a year ago, I first heard about ReactOS from a friend of mine when I traveling with him.

We try to install ReactOS on a VM to see what exactly is it. Unfortunately, it crashed when installing and a blue screen was shown. But anyway that's my first meet with ReactOS. I didn't realize that I would work on this project in the future so soon.

Several months ago, I first heard about GSoC (Interestingly, still from the same friend). When I try to find a project correspond with my skill, ReactOS came into my view.

It's great for me because having a chance to write a part of an OS is exciting and there aren't many projects that require Win32 skills. And then I'm here!

## Details about the project

RAPPS is a lightweight GUI utility that downloads and installs various popular apps and redistributable packages in Windows and ReactOS. Aiming at making manage software easier, especially on ReactOS.

> Currently, RAPPS gots dozens of bugs to be fixed, and some function is broken or not implemented. besides, I consider the UI of RAPPS is a little bit old-fashioned (this may differ from man to man). My aim is to fix bugs as much as I can and improve RAPPS's functionality and UI.

Reviewer: ```suggestion
Currently, RAPPS has dozens of bugs to be fixed, and some function are broken or not implemented.
Besides, I consider the UI of RAPPS is a little bit old-fashioned (this may differ from man to man).
My aim is to fix bugs as much as I can and improve RAPPS's functionality and UI.
```

> During the community bonding period, I spend some time to get me familiar with the code, I've already done some simple work now (for example, move database source URL to settings dialog, see PR 2792)

Reviewer: ```suggestion
During the community bonding period, I spent some time to get familiar with the code.
I've already done some simple work (for example, move database source URL to settings dialog, see PR 2792)
```

### Current progress

> What I'm going to do recently is support software snapshot view and better support for icon display. After this is done, I will refactor that list view and support display software as icon/list/detail. 

Reviewer: ```suggestion
What I'm going to do soon is support software snapshot view and better support for icon display.
After this is done, I will refactor that list view and support display software as icon/list/detail. 
```

### Coming future plan

> I will also improve command-line support, and improve current download (allow download more at the same time, etc.)

Reviewer: ```suggestion
I will also improve command-line support, and improve concurrent download (allow download more at the same time, etc.)
```

> If I still have time left(which I guess I won't), I could add more features, and If you got any good idea, feel free to discuss with me!

Reviewer: ```suggestion
If I still have time left (which I guess I won't), I could add more features, and If you got any good idea, feel free to discuss with me!
```

---
title:       "Rapps Enhancement - GSoC 2020"
author:      "kernel.bin"

> date:        2020-05-30

Reviewer: ```suggestion
date:        2020-06-04
```

> tags:        [ gsoc ]

Reviewer: ```suggestion
tags:        [ "gsoc", "gsoc-2020" ]
```
